gflownet/envs/ctorus.py

```diff
@@ -72,34 +80,56 @@ def get_policy_output(self, params: dict):
 
     def get_mask_invalid_actions_forward(self, state=None, done=None):
         """
-        Returns [True] if the only possible action is eos, [False] otherwise.
+        The action is space is continuous, thus the mask is not of invalid actions as
+        in discrete environments, but an indicator of "special cases", for example
+        states from which only certain actions are possible.
+
+        The "mask" has 2 elements - to match the mask of backward actions - but only
+        one is needed for forward actions, thus both elements take the same value,
+        according to the following:
+
+        - If done is True, then the mask is True.
+        - If the number of actions (state[-1]) is equal to the (fixed) trajectory
+          length, then only EOS is valid and the mask is True.
+        - Otherwise, any continuous action is valid (except EOS) and the mask is False.
```

Comment:
I get that this mask is in a way fake, but for consistency wouldn't it make more sense to set the first entry of the mask (corresponding to the generic action) to True, and the second (corresponding to EOS) to False if the number of actions is equal to the trajectory length, and the other way around if any continuous action is valid (except EOS) is valid?

Reply:
Yes that would make sense actually, but may need changes in `sample_actions_batch` and `get_logprobs`

Reply:
I am making this an issue instead of changing now because things work fine anyway and that implies changing + testing + re-reviewing, etc.

Reply:
Hope you don't mind.